vm/systemSmartContracts/auction.go

```diff
@@ -925,40 +925,71 @@ func (s *stakingAuctionSC) unStakeNodeOneNodeFromStakingSC(blsKey []byte, reward
 	return vmOutput.ReturnCode
 }
 
-func (s *stakingAuctionSC) unBond(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+func (s *stakingAuctionSC) unBondNodesOnly(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !s.flagEnableTopUp.IsSet() {
+		s.eei.AddReturnMessage("invalid method to call")
+		return vmcommon.UserError
+	}
+	registrationData, returnCode := s.checkUnBondArguments(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+
+	totalUnBond, unBondedKeys := s.unBondNodesFromStakingSC(args.Arguments)
+	if registrationData.NumRegistered < uint32(len(unBondedKeys)) {
+		s.eei.AddReturnMessage("contract error on unBond function, missing nodes")
+		return vmcommon.UserError
+	}
+
+	registrationData.NumRegistered -= uint32(len(unBondedKeys))
+	registrationData.TotalStakeValue.Add(registrationData.TotalStakeValue, totalUnBond)
```

Comment:
This will suffer some changes in the next PR. This PR was more about delegation SC and to make the unit test pass. The next will focus on auction. After that one more on staking

Reply:
ok